tests/crustal_model_3D.prm

```diff
@@ -0,0 +1,131 @@
+
+############### Global parameters
+
+set Dimension                              = 3
+set Start time                             = 0
+set End time                               = 0
+set Use years in output instead of seconds = true
+set CFL number                             = .5
+set Linear solver tolerance                = 1e-9
+set Nonlinear solver scheme                = iterated Stokes
+set Nonlinear solver tolerance             = 1e-5
+set Max nonlinear iterations               = 3
+set Number of cheap Stokes solver steps    = 0
+set Pressure normalization                 = no
+set Timing output frequency                = 1
+
+############### Parameters describing the model
+
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X extent  = 128e3
+    set Y extent  =  96e3
+    set Z extent  =  16e3
+    set X repetitions  = 8
+    set Y repetitions  = 3
+  end
+end
+
+subsection Model settings
+  set Tangential velocity boundary indicators = front,back 
+  set Prescribed velocity boundary indicators = left: function, right:function, bottom:function
+  set Free surface boundary indicators        = top
+end
+
+
+# Advecting the free surface vertically rather than
+# in the surface normal direction can result in a
+# more stable mesh when the deformation is large
+subsection Free surface
+  set Surface velocity projection = vertical
+end
+
+
+subsection Material model
+  set Model name = drucker prager
+  subsection Drucker Prager
+    set Reference density = 2800
+    subsection Viscosity
+       set Minimum viscosity = 1e19
+       set Maximum viscosity = 1e25 
+       set Reference strain rate = 1e-20
+       set Angle of internal friction = 30
+       set Cohesion = 20e6 
+    end
+  end
+end
+
+subsection Gravity model
+  set Model name = vertical
+  subsection Vertical
+    set Magnitude = 9.81
+  end
+end
+
+
+
+# The velocity along the top boundary models a spreading
+# center that is moving left and right:
+subsection Boundary velocity model
+  subsection Function
+    set Variable names      = x,y,z
+    set Function constants  = cm=0.01, year=1 
+    set Function expression =  if (x<56e3 && y<=48e3 | x<72e3 && y>48e3, -1*cm/year, 1*cm/year) ; 0 ; 0
+  end
+end
+
```

Comment:
One more comment - the velocity is applied to the to left and right walls in this case. In your comment above are you referring to the velocity that arises along the top boundary due to the applied velocities along the left and right walls?